cmd/kubectl-gadget/collector.go

```diff
@@ -0,0 +1,389 @@
+				VersionedParams(&listTracesOptions, scheme.ParameterCodec).
+				Do(context.TODO()).
+				Into(&results)
+			if err != nil {
+				contextLogger.Fatalf("Error getting traces: %q", err)
+			}
+			again := false
+			for _, i := range results.Items {
+				if i.Status.State != "Completed" {
+					again = true
+				}
+			}
+			if again {
+				elapsed := time.Now().Sub(start)
+				if elapsed > 2*time.Second {
+					contextLogger.Fatalf("Error waiting for traces: timeout: %v", elapsed)
+				}
+				time.Sleep(200 * time.Millisecond)
+				continue
+			} else {
+				break
+			}
+		}
+
+		// Delete traces in one call
+		deleteResults := gadgetv1alpha1.TraceList{}
+		var deleteTracesOptions = metav1.ListOptions{
+			LabelSelector: fmt.Sprintf("trace-template-hash=%s", traceID),
+			FieldSelector: fields.Everything().String(),
+		}
+		err = traceRestClient.
+			Delete().
+			Namespace("gadget").
+			Resource("traces").
+			VersionedParams(&deleteTracesOptions, scheme.ParameterCodec).
+			Do(context.TODO()).
+			Into(&deleteResults)
+		if err != nil {
+			contextLogger.Fatalf("Error deleting traces: %q", err)
+		}
+
+		// Display results
+		type Process struct {
+			Tgid                int    `json:"tgid,omitempty"`
+			Pid                 int    `json:"pid,omitempty"`
+			Comm                string `json:"comm,omitempty"`
+			KubernetesNamespace string `json:"kubernetes_namespace,omitempty"`
+			KubernetesPod       string `json:"kubernetes_pod,omitempty"`
+			KubernetesContainer string `json:"kubernetes_container,omitempty"`
+		}
+		var allProcesses []Process
+
+		for _, i := range results.Items {
+			var processes []Process
+			json.Unmarshal([]byte(i.Status.Output), &processes)
+			allProcesses = append(allProcesses, processes...)
+		}
+		if !collectorParamThreads {
+			var allProcessesTrimmed []Process
+			for _, i := range allProcesses {
+				if i.Tgid == i.Pid {
+					allProcessesTrimmed = append(allProcessesTrimmed, i)
+				}
+			}
+			allProcesses = allProcessesTrimmed
+		}
+
+		sort.Slice(allProcesses, func(i, j int) bool {
+			pi, pj := allProcesses[i], allProcesses[j]
+			switch {
+			case pi.KubernetesNamespace != pj.KubernetesNamespace:
+				return pi.KubernetesNamespace < pj.KubernetesNamespace
+			case pi.KubernetesPod != pj.KubernetesPod:
+				return pi.KubernetesPod < pj.KubernetesPod
+			case pi.KubernetesContainer != pj.KubernetesContainer:
+				return pi.KubernetesContainer < pj.KubernetesContainer
+			case pi.Comm != pj.Comm:
+				return pi.Comm < pj.Comm
+			case pi.Tgid != pj.Tgid:
+				return pi.Tgid < pj.Tgid
+			default:
+				return pi.Pid < pj.Pid
+
+			}
+		})
+		if collectorParamJsonOutput {
+			b, err := json.MarshalIndent(allProcesses, "", "  ")
+			if err != nil {
+				contextLogger.Fatalf("Error marshalling results: %s", err)
+			}
+			fmt.Printf("%s\n", b)
+		} else {
+			w := tabwriter.NewWriter(os.Stdout, 0, 0, 4, ' ', 0)
+			if collectorParamThreads {
+				fmt.Fprintln(w, "NAMESPACE\tPOD\tCONTAINER\tCOMM\tTGID\tPID\t")
```

Comment:
Is the node missing?

Reply:
Indeed, the process-collector gadget does not get the node... It can be added in another PR in the same way you did for other bcc gadgets.